src/multi-caching.ts

```diff
@@ -25,12 +25,23 @@ export function multiCaching<Caches extends Cache[]>(
     del: async (key) => {
       await Promise.all(caches.map((cache) => cache.del(key)));
     },
-    async wrap<T>(key: string, fn: () => Promise<T>): Promise<T> {
-      const value = await get<T>(key);
+    async wrap<T>(key: string, fn: () => Promise<T>, ttl?: Ttl): Promise<T> {
+      let value: T | undefined;
+      let i = 0;
+      for (; i < caches.length; i++) {
+        try {
+          value = await caches[i].get<T>(key);
+          if (value !== undefined) break;
+        } catch (e) {}
+      }
       if (value === undefined) {
         const result = await fn();
-        await set<T>(key, result);
+        await set<T>(key, result, ttl);
         return result;
+      } else {
+        for (let j = 0; j < i; j++) {
+          await caches[j].set(key, value, ttl);
+        }
```

Comment:
I guess that this could be done in parallel and we could avoid waiting for it to complete, like it was before afair.
```suggestion
for (let j = 0; j < i; j++) {
  caches[j].set(key, value, ttl).catch(() => {});
}
```